cve_bin_tool/version_scanner.py

```diff
@@ -160,13 +163,86 @@ def scan_file(self, filename):
         # parse binary file's strings
         lines = self.parse_strings(filename)
 
+        # Check for Java package
+        if output and "pom.xml" in output:
+            java_lines = "\n".join(lines.splitlines())
+            yield from self.run_java_checker(filename, java_lines)
+
         #  If python package then strip the lines to avoid detecting other product strings
         if output and ("PKG-INFO: " in output or "METADATA: " in output):
             py_lines = "\n".join(lines.splitlines()[:3])
             yield from self.run_python_package_checkers(filename, py_lines)
 
         yield from self.run_checkers(filename, lines)
 
+    def find_java_vendor(self, product, version):
+        """Find vendor for Java product"""
+        vendor_package_pair = self.cve_db.get_vendor_product_pairs(product)
+        # If no match, try alternative product name.
+        # Apache product names are stored as A_B in NVD database but often called A-B
+        # Some packages have -parent appended to product which is not in NVD database
+        if vendor_package_pair == [] and "-" in product:
+            self.logger.debug(f"Try alternative product {product}")
+            # Remove parent appendage
+            if "-parent" in product:
+                product = product.replace("-parent", "")
+            product = product.replace("-", "_")
+            vendor_package_pair = self.cve_db.get_vendor_product_pairs(product)
+        if vendor_package_pair != []:
+            vendor = vendor_package_pair[0]["vendor"]
+            file_path = "".join(self.file_stack)
+            self.logger.debug(f"{file_path} {product} {version} by {vendor}")
+            return ProductInfo(vendor, product, version), file_path
+        return None, None
+
+    def run_java_checker(self, filename, lines):
+        """Process maven pom.xml file and extract product and dependency details"""
+        tree = ET.parse(filename)
```

Comment:
Does maven have any sort of xmlschema we should be using to verify that the data is valid?  I feel like it *should* have a published one we could use (and possibly store locally in our source tree to avoid the network connection), but I won't be surprised if that's not the case.